WLC/code_fixing/codefixer.py

```diff
@@ -1,92 +1,181 @@
             fixed_lines.append(new_line)
 
         return "\n".join("{indent}{code}".format(indent="  " * indent, code=line) for indent, line in
                          zip(self.indents, fixed_lines))
 
-    def _fix_word(self, word, contextual_data=None, prev_context=""):
-        if contextual_data is None:
-            contextual_data = []
-
-        if prev_context == "import":  # no numbers -> replace possibly similar matches.
-            # word = "".join(character.get_code(get_letters=True).lower() for character in characters)
-            # TODO: convert ints to chars by looking at predictions
-            contextual_data.append(word)  # add the imported module to contextual
-            return word, contextual_data
-
-        elif not prev_context:
-            start = 0
-            end = 0
-            l_dist = 10  # max change is 2 anyway
-            curr_best_match = None
-            curr_best_word = None
-
-            for kw in contextual_data + KW_LIST:
-                pos = find_near_matches(kw, word, max_l_dist=min(2, len(kw) - 1), max_insertions=0, max_deletions=0)
-
-                for p in pos or []:
-                    if p.start == 0:
-                        # If word is longer, or if theres less l_dist (prioritising contextual_data)
-                        if end < p.end or (end == p.end and p.dist < l_dist and not (
-                                curr_best_word in contextual_data and kw in KW_LIST)):
-                            end = p.end
-                            curr_best_match = p
-                            curr_best_word = kw
-                            l_dist = p.dist
-
-            if curr_best_match and curr_best_word:
-                new_word = word[:start] + curr_best_word + word[end:]
-                word = new_word
-
-        return word, contextual_data
-
-    def _fix_line(self, line, contextual_data=None):
-        if contextual_data is None:
-            contextual_data = []
-
+    def _levenshtein(self, word, check_on=None):
+        if not check_on:
+            check_on = self.context + KW_LIST
+
+        start = 0  # doesn't change
+        end = 0
+        l_dist = 10  # max change is 2 anyway
+        curr_best_word = None
+
+        for kw in check_on:
+            pos = find_near_matches(kw, word, max_l_dist=min(2, len(kw) - 1), max_insertions=0, max_deletions=0)
+
+            for p in pos or []:
+                if p.start == start:
+                    # If word is longer, or if theres less l_dist (prioritising contextual_data)
+                    if end < p.end or (end == p.end and p.dist < l_dist and not (
+                                    curr_best_word in self.context and kw in KW_LIST)):
+                        end = p.end
+                        curr_best_word = kw
+                        l_dist = p.dist
+
+        return curr_best_word, start, end
+
+    def _fix_word(self, word, word_number=None, line_number=None, prev_word="", next_word=""):
```

Comment:
`word_number` and `line_number` not used

Reply:
They're  placeholders for when we start bruteforcing